src/index.js

```diff
@@ -0,0 +1,129 @@
+  const minCelsius = ((Number(minTemperature) - 32) * (5 / 9)).toFixed(2);
+
+  const maxTemperature = maxTemp.textContent.split(' ')[3].split('°')[0];
+  const maxFahrenheit = ((Number(maxTemperature) * (9 / 5)) + 32).toFixed(2);
+  const maxCelsius = ((Number(maxTemperature) - 32) * (5 / 9)).toFixed(2);
+
+  currentTemp.innerHTML = toggleButtonValue.checked === false ? `${celsius} °<span>C</span>` : `${fahrenheit} °<span>F</span>`;
+  minTemp.innerHTML = toggleButtonValue.checked === false ? `Min Temp :- ${minCelsius}°<span>C</span> ` : `Min Temp :- ${minFahrenheit}°<span>F</span> `;
+  maxTemp.innerHTML = toggleButtonValue.checked === false ? `Max Temp :- ${maxCelsius}°<span>C</span> ` : `Max Temp :- ${maxFahrenheit}°<span>F</span> `;
+};
+
+const updateData = (data) => {
+  const celsius = data.main.temp;
+  const windSpeedConvert = Math.floor(data.wind.speed);
+
+  location.textContent = data.name;
+  currentTemp.innerHTML = `${celsius} °<span>C</span>`;
+  minTemp.innerHTML = `Min Temp :- ${data.main.temp_min}°<span>C</span> `;
+  maxTemp.innerHTML = `Max Temp :- ${data.main.temp_max}°<span>C</span> `;
+  weatherReport.textContent = data.weather[0].description;
+  humidity.innerHTML = `Humidity :- ${data.main.humidity}% `;
+  windSpeed.innerHTML = `Wind Speed :- ${windSpeedConvert}<span>Km/hr</span> `;
+};
+
+toggleTemperatureButton.addEventListener('change', (e) => {
+  toggleButtonValue.checked = e.target.checked;
+  convertTemp();
+});
+
+window.addEventListener('load', () => {
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(async (position) => {
+      const searchCity = document.querySelector('form');
+      const lat = position.coords.latitude;
+      const long = position.coords.longitude;
+
+      const api = await fetch(`https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${long}&appid=${key}&units=metric`);
+      const data = await api.json();
+
+      searchCity.addEventListener('keypress', async (e) => {
+        try {
+          if (e.keyCode === 13) {
+            e.preventDefault();
+            const searchValue = e.target.value;
+            if (searchValue !== '') {
+              const searchResult = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=${searchValue}&appid=${key}&units=metric`);
+              const data = await searchResult.json();
+
+              updateData(data);
+              icon.remove('icon9');
+              icon.set('icon9', iconWeatherCheck(data.weather[0]));
+              placeholderValue.classList.remove('error');
+              placeholderValue.placeholder = 'Enter a city name';
+              e.target.value = '';
+            }
+          }
+        } catch (err) {
+          e.target.value = '';
+          placeholderValue.placeholder = 'Enter a valid city name';
+          placeholderValue.classList.add('error');
+        }
+      });
+
+      updateData(data);
+      icon.add('icon9', iconWeatherCheck(data.weather[0]));
+    });
```

Comment:
- we are required to not mix LOGIC and DOM in this project
The weather-fetching-code appears to be the main logic here, look for a way to get that logic in a separate file, such that it takes a certain city in parameter and returns whether information. You can now use this function in a DOM module to play with the data

[ see [requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/dom_logic_separation_requirement.png)]

Reply:
Hey Shloch, thanks for the suggestion, now I can see my code base is even cleaner. And thanks for your feedback on the project.